aten/src/ATen/native/SharedReduceOps.h

```diff
@@ -103,6 +103,11 @@ struct WelfordOps {
   inline C10_DEVICE res_t project(acc_t acc) const {
     auto mean = acc.mean;
     combine_t divisor = unbiased ? (acc.nf - 1) : acc.nf;
+    if (acc.nf == 1) {
+        // one observation, variance is 0
+        detail::pair<scalar_t, scalar_t> results{(scalar_t) 0, (scalar_t) mean};
+        return results;
```

Comment:
The NumPy result for 0 d.o.f is infinity BTW. So this is in disagreement.

Also, from the mathematical perspective an unbiased estimate of the population variance cannot be determined from a single sample. So, I think NumPy is reasonable in this regard. I could see there is an argument for nf=0 though.

Reply:
> The NumPy result for 0 d.o.f is infinity BTW.

If the population has one value the unbiased estimate (0 d.o.f.) is Nan not Inf. The biased estimate (1 d.o.f.) is 0. The default for numpy is different from Pytorch: numpy gives a biased estimate by default and pytorch gives an unbiased by default. That is what was confusing me.

Reply:
>If the population has one value the unbiased estimate (0 d.o.f.) is Nan not Inf. The biased estimate (1 d.o.f.) is 0. 

So we're on the same wavelength. The degrees of freedom (d.o.f.) is `N - ddof` where NumPy's `ddof` is equal to 1 for Bessel's correction (unbiased estimator).

Actually, it looks like NumPy isn't completely consistent here:
```
np.var([1.0] * 100, ddof=100) -> nan
np.var(np.random.rand(100), ddof=100) -> inf
```
I think this is down to the numerator. `0/0 = nan` but `positive / 0 = inf`. Either way, when there are no degrees of freedom, the variance cannot be finite.